File: frontend/src/features/auth/dto/session-token-response.ts
import type { PermissionSnapshot } from "./permission-snapshot"

/**
 * Returned when a session is created — by both `mfaVerify` and
 * `mfaEnrollConfirm`.
 */
export interface SessionTokenResponse {
  sessionToken: string
  userId: string
  expiresAtUtc: string
  permissionSnapshot: PermissionSnapshot
}

File: src/Nabadat.UserManagement/Api/Contracts/SessionTokenResponse.cs
using Nabadat.UserManagement.Domain.ValueObjects;

namespace Nabadat.UserManagement.Api.Contracts;

/// <summary>Session-created response shared by MFA verify and enrollment confirm.</summary>
public sealed record SessionTokenResponse
{
    public required string SessionToken { get; init; }

    public required Guid UserId { get; init; }

    public required DateTimeOffset ExpiresAtUtc { get; init; }

    public required PermissionSnapshot PermissionSnapshot { get; init; }
}
